cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -2964,6 +3138,24 @@ bool CacheAllocator<CacheTrait>::markMovingForSlabRelease(
     if (itemFreed) {
       return false;
     } else if (markedMoving) {
+      Item* item = static_cast<Item*>(alloc);
+      XDCHECK(syncItem->isMoving());
+      XDCHECK(item->isChainedItem()
+              ? item->asChainedItem().getParentItem(compressor_).isMoving()
+              : item->isMoving()) << item->toString() << "\n" << syncItem->toString();
+      if ( ( item->isChainedItem() &&
```

Comment:
When can this happen?

Reply:
I think just `XDCHECK(syncItem->isMoving());` is enough. All other checks just dup;lications.